experiments/hello-network-socket-ping/local_pong.c

```diff
@@ -0,0 +1,212 @@
+#include <errno.h>
+#include <stdio.h>
+#include <unistd.h>
+
+#include <sys/un.h>
+#include <sys/socket.h>
+
+#include "shared.h"
+#include "util.h"
+
+#define SIZE_RECEIVE_BUFFER 512
+
+const char* g_program_name = "Pong";
+
+// TODO
+// * write separate files for Unix domain sockets and internet domain sockets?
+
+// TODO consider rename to: prepare_for_bind_server
+static int
+prepare_for_bind(struct sockaddr_un* addr_ptr, const char* sock_path)
+{
+    char error_message[SIZE_ERROR_MESSAGE];
+    int32_t ret;
+
+    ret = 0;
+    memset(error_message, 0, SIZE_ERROR_MESSAGE);
+
+    if (strlen(sock_path) > sizeof(addr_ptr->sun_path) - 1) {
+        sscanf(
+                error_message,
+                "Socket path is too long (%d characters)",
+                strlen(sock_path));
+        error_message[255] = '\0';
+        print_error(error_message);
+        return -1;
+    }
+
+    // Ensure there is no pre-existing file with wanted name
+    ret = remove(sock_path);
+    if (-1 == ret && ENOENT != errno) {
+        perror("Failed to remove file");
+        sscanf(
+                error_message,
+                "Failed to remove file, using path: %s",
+                sock_path);
+        error_message[255] = '\0';
+        print_error(error_message);
+        return -1;
+    }
+
+    memset(addr_ptr, 0, sizeof(struct sockaddr_un));
+
+    addr_ptr->sun_family = AF_UNIX;
+
+    // Using sizeof the whole struct had two bytes (short) extra when compared
+    // to using sizeof sockaddr_un.sun_path. Checked from system header files.
+    strncpy(addr_ptr->sun_path, sock_path, sizeof(struct sockaddr_un) - 1);
+
+    return 0;
+}
+
+static int
+receive_message(
+        int32_t const * const sock_fd_ptr,
+        struct sockaddr_un* address_client_ptr,
+        char* out_message)
+{
+    socklen_t len;
+    ssize_t bytes_received;
+
+    char receive_buffer[SIZE_RECEIVE_BUFFER];
+
+    len = 0;
+    bytes_received = 0;
+    memset(receive_buffer, 0, SIZE_RECEIVE_BUFFER);
+
+    len = sizeof(struct sockaddr_un);
+    bytes_received = recvfrom(
+            *(sock_fd_ptr),
+            receive_buffer,
+            SIZE_RECEIVE_BUFFER,
+            0,
```

Comment:
Please use a variable instead of magic 0.

Reply:
Done in: https://github.com/arbiter0xf/neotetris/pull/8/commits/4a6e67db7dd739e0d57c804041f6c0642d806b7d